src/main/java/org/broadinstitute/hellbender/tools/ConvertHeaderlessHadoopBamShardToBam.java

```diff
@@ -1,44 +1,78 @@
 package org.broadinstitute.hellbender.tools;
 
-import htsjdk.samtools.*;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SamReader;
+import htsjdk.samtools.SamReaderFactory;
+import htsjdk.samtools.ValidationStringency;
 import org.broadinstitute.barclay.argparser.Argument;
 import org.broadinstitute.barclay.argparser.BetaFeature;
 import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
 import org.broadinstitute.barclay.help.DocumentedFeature;
 import org.broadinstitute.hellbender.cmdline.CommandLineProgram;
-import org.broadinstitute.hellbender.cmdline.programgroups.TestSparkProgramGroup;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.ReadDataProgramGroup;
 import org.broadinstitute.hellbender.engine.spark.datasources.ReadsSparkSink;
 import org.broadinstitute.hellbender.exceptions.UserException;
 import org.broadinstitute.hellbender.utils.read.ReadsWriteFormat;
 import org.broadinstitute.hellbender.utils.spark.SparkUtils;
 
-import java.io.*;
+import java.io.File;
+import java.io.IOException;
 
 
 /**
- * Spark troubleshooting utility that converts a headerless Hadoop bam shard (eg., a part0000, part0001, etc. file
- * produced by {@link ReadsSparkSink}) into a readable bam file by adding a header and a BGZF terminator.
+ * Convert a headerless Hadoop BAM shard into a readable BAM. This is a Spark troubleshooting utility that converts a
+ * headerless Hadoop BAM shard (eg., a part-r-00000, part-r-00001, etc. file
+ * produced by a Spark tool using {@link ReadsSparkSink}) into a readable BAM file by adding a header and a BGZF terminator.</p>
  *
- * This tool is not intended for use with Hadoop bam shards that already have a header -- these shards are
+ * <p>This tool is not intended for use with Hadoop bam shards that already have a header -- these shards are
  * already readable using samtools. Currently {@link ReadsSparkSink} saves the "shards" with a header for the
- * {@link ReadsWriteFormat#SHARDED} case, and without a header for the {@link ReadsWriteFormat#SINGLE} case.
+ * {@link ReadsWriteFormat#SHARDED} case, and without a header for the {@link ReadsWriteFormat#SINGLE} case.</p>
+ *
+ * <h3>Input</h3>
+ * <ul>
+ *     <li>A BAM shard</li>
+ *     <li>A well-formed BAM whose header will be used for the converted fragment</li>
+ * </ul>
+ *
+ * <h3>Output</h3>
+ * <ul>
+ *     <li>The converted BAM shard</li>
+ * </ul>
+ *
+ * <h3>Usage example</h3>
+ * <pre>
+ * gatk ConvertHeaderlessHadoopBamShardToBam \
+ *     --bam-shard shard.bam \
+ *     --bam-with-header input.bam \
+ *     --O output.bam
```

Comment:
This example seems to imply you can use a file that is already in bam format (has header and terminator). I tested this command as you show and the command runs without error but produces an invalid output (that samtools refuses to display). Also `--O`-->`-O`.

Should this example access files that represent a Hadoop structure as mentioned above, e.g. with something like: 
```
gatk ConvertHeaderlessHadoopBamShardToBam \
     --bam-shard part-r-00001 \
     --bam-with-header input.bam \
     -O part-r-00001.bam
 ```
Finally, could the bam-with-header be instead a reference dictionary? I think yes as I tested this to create a similarly truncated file that my setup produces. So add another example command to the Usage examples:

Use a reference dictionary file as the header:
```
gatk ConvertHeaderlessHadoopBamShardToBam \
     --bam-shard part-r-00001 \
     --bam-with-header reference.dict \
     -O part-r-00001.bam
```

Hopefully from a real shard part this produces a nontruncated full BAM.

Reply:
Fixed the -O argument. I've changed the shard name to part-r-00000.bam to be consistent with the current shard naming convention:

```gatk ConvertHeaderlessHadoopBamShardToBam \
     --bam-shard part-r-00000.bam \
     --bam-with-header input.bam \
     -O  output.bam```

I'm hesitant to add a dictionary example to the doc because the code uses a SAM reader from htsjdk that expects a SAM/BAM file. It happens to work with a dictionary at the moment, but this is semantically different from the intended use of the reader.